.github/workflows/push.yml

```
@@ -0,0 +1,13 @@
+name: Pre-commit hooks
+
+on: [push]
+
+jobs:
+  pre-commit:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-python@v5
+      - uses: pre-commit/action@v3
```

```suggestion
      - uses: pre-commit/action@v3.0.1
```

See https://github.com/pre-commit/action/issues/104 and https://github.com/pre-commit/action/issues/79.

---

@francois-rozet I gave that review on purpose :) pre-commit/action is not maintained anymore and it seems there will only be performance / caching upgrades from now on. Not having it pinned down to patch version doesn't create a functional risk, and having it pinned creates extra work to fetch those updates.

---

`@v3` does not work. GitHub action  versions can be loose if the repo allows it. The maintainer of pre-commit is against loose versions (see the issues I linked).

Still, I also dislike this pinned version, so we might as well replace it with manually calling pre-commit. Accepting PRs ;)